### PR TITLE
tests: query testing

### PR DESCRIPTION
**Pull request by dinosk**
Signed-off-by: Dinos Kousidis <konstantinos.kousidis@cern.ch>

### CODE REVIEW

File: tests/test_queries.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+             (2017, 1, 7)]
+    events = [_create_file_download_event(date) for date in dates]
+    current_queues.declare()
+    current_stats.publish(event_type, events)
+    process_events(['file-download'])
+    current_search_client.indices.flush(index='*')
+    aggregate_events(['file-download-agg'])
+    current_search_client.indices.flush(index='*')
+
+    query_configs = register_queries()
```
Comment (nharraud) [inline tests/test_queries.py]: Add a comment to explain.

File: tests/test_queries.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    dates = [(2017, 1, 1),
+             (2017, 1, 2),
+             (2017, 1, 3),
+             (2017, 1, 3, 5),
+             (2017, 1, 4),
+             (2017, 1, 5),
+             (2017, 1, 6),
+             (2017, 1, 7)]
+    events = [_create_file_download_event(date) for date in dates]
+    current_queues.declare()
```
Comment (nharraud) [inline tests/test_queries.py]: Should already be done by `event_queues`

File: tests/test_queries.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+    query_configs = register_queries()
+    histo_query = ESDateHistogramQuery(query_name='test_histo',
+                                       **query_configs[0]['query_config'])
+    results = histo_query.run(bucket_id='B0000000000000000000000000000001',
+                              file_key='test.pdf',
+                              start_date=datetime.datetime(2017, 1, 1),
+                              end_date=datetime.datetime(2017, 1, 7))
+    for idx, day_result in enumerate(results['buckets']):
+        if idx == 2:
```
Comment (nharraud) [inline tests/test_queries.py]: a comment to make it clear.

File: tests/test_queries.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+            assert int(day_result['value']) == 2
+        else:
+            assert int(day_result['value']) == 1
+
+    terms_query = ESTermsQuery(query_name='test_total_count',
```
Comment (nharraud) [inline tests/test_queries.py]: This should be in another test (test_terms_query)

File: tests/test_queries.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+             (2017, 1, 2),
+             (2017, 1, 3),
+             (2017, 1, 3, 5),
+             (2017, 1, 4),
+             (2017, 1, 5),
+             (2017, 1, 6),
+             (2017, 1, 7)]
+    events = [_create_file_download_event(date) for date in dates]
+    current_queues.declare()
+    current_stats.publish(event_type, events)
```
Comment (nharraud) [inline tests/test_queries.py]: You don't need to really create the events. You can mock it.

File: tests/conftest.py [hunk 6]
```diff
@@ -137,7 +199,8 @@ def event_queues(app, event_entrypoints):
 def base_app():
     """Flask application fixture without InvenioStats."""
     from invenio_stats.config import STATS_EVENTS
-    app_ = Flask('testapp')
+    instance_path = tempfile.mkdtemp()
```
Comment (nharraud) [inline tests/conftest.py]: The instance directory is not removed at the end.

Comment (dinosk) [reply tests/conftest.py]: ah right, fixed.

**Pull request merged**